Common/Mpu6050.hpp

```diff
@@ -0,0 +1,196 @@
+  ///        information. Where as 2g has a lower maximum detectable
+  ///        acceleration, but 12 bits of information.
+  /// @param address - Mpu6050 device address.
+  explicit constexpr Mpu6050(
+      const I2c & i2c,
+      units::acceleration::standard_gravity_t full_scale = 2_SG,
+      uint8_t address                                    = 0x68)
+      : i2c_(i2c), kFullScale(full_scale), kAccelerometerAddress(address)
+  {
+  }
+
+  Returns<void> Initialize() override
+  {
+    Status status = i2c_.Initialize();
+    i2c_.Write(kAccelerometerAddress, { 0x6B, 0x0 });
+
+    // TODO(#1244): Migrate to using auto return macro SJ2_RETURN_VALUE_ON_ERROR
+    if (!IsOk(status))
+    {
+      return Error(Status::kBusError, "I2C Initialize Failed!");
+    }
+
+    return {};
+  }
+
+  /// Will automatically set the device to 2g for full-scale. Use
+  /// `SetFullScaleRange()` to change it.
+  Returns<void> Enable() override
+  {
+    // Check that the device is valid before proceeding.
+    SJ2_RETURN_ON_ERROR(IsValidDevice());
+
+    // Put device into standby so we can configure the device.
+    // SJ2_RETURN_ON_ERROR(ActiveMode(false));
+
+    // Set device full-scale to the value supplied by the constructor.
+    SJ2_RETURN_ON_ERROR(SetFullScaleRange());
+
+    // Activate device to allow full-scale and configuration to take effect.
+    // SJ2_RETURN_ON_ERROR(ActiveMode(true));
+
+    return {};
+  }
+
+  Returns<Acceleration_t> Read() override
+  {
+    constexpr uint16_t kBytesPerAxis = 2;
+    constexpr uint8_t kNumberOfAxis  = 3;
+
+    Acceleration_t acceleration = {};
+
+    uint8_t xyz_data[kBytesPerAxis * kNumberOfAxis];
+
+    i2c_.WriteThenRead(kAccelerometerAddress,
+                       { Value(RegisterMap::kXYZStartAddress) }, xyz_data,
+                       sizeof(xyz_data));
+
+    // First X-axis Byte (MSB first)
+    // =========================================================================
+    // Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+    //  XD15 | XD14  |  XD13 |  XD12 |  XD11 |  XD10 |  XD9  |  XD8
+    //
+    // Final X-axis Byte (LSB)
+    // =========================================================================
+    // Bit 7 | Bit 6 | Bit 5 | Bit 4 | Bit 3 | Bit 2 | Bit 1 | Bit 0
+    //   XD7 |   XD6 |   XD5 |   XD4 |   XD3 |   XD2 |   XD1 |   XD0
+    //
+    // We simply shift and OR the bytes together to get them into a signed int
+    // 16 value.
+
+    int16_t x = static_cast<int16_t>(xyz_data[0] << 8 | xyz_data[1]);
+    int16_t y = static_cast<int16_t>(xyz_data[2] << 8 | xyz_data[3]);
+    int16_t z = static_cast<int16_t>(xyz_data[4] << 8 | xyz_data[5]);
+
+    //Convert the 16 bit value into a floating point value in SG
+    acceleration.x = kFullScale * (float)x / (INT16_MAX+1);
+    acceleration.y = kFullScale * (float)y / (INT16_MAX+1);
+    acceleration.z = kFullScale * (float)z / (INT16_MAX+1);
```

Comment:
Tip #1 use `static_cast` for casting. Do NOT use C style casting. It is an unsafe operation.
Tip #2 refrain from using macros in C++ when C++ alternatives exists. For example

```C++
#include <limits>
// ...
int16_t max = std::numeric_limits<int16_t>::max();
```

Tip #3 use the map function here to map the input values to the output values. 

```suggestion
    constexpr int16_t kMax = std::numeric_limits<int16_t>::max();
    constexpr int16_t kMin = std::numeric_limits<int16_t>::min();
    
    float x_ratio = sjsu::Map(x, kMin, kMax, -1.0f, 1.0f);
    acceleration.x = kFullScale * x_ratio;
    // Do the same for the other axis
```